### ISSUE TITLE
Datadog Direct to Datadog API Approach

### ISSUE THREAD

**Issue by paulohenriqu** [micrometer-metrics/micrometer#3911]
**Please describe the feature request.**
Send metrics to the Datadog API with the same format the datadog agent uses. More specifically sending counts as rate metrics.

**Rationale**
f you've created a metric/custom metric as a count using Datadog's agent/dogstatsd, it will have the type of Rate interval=10 in Datadog by default. Not only does it transmit the values this way, it actually divides the values provided by 10 in the data series -- https://docs.datadoghq.com/api/latest/metrics/#submit-metrics.
So when you submit the same metric using the micrometer integration direct to the API the values are multiplied by 10.

It would be great to have a configuration to send metrics using Micrometer with the same format as the Datadog agent sends.

**Additional context**
Very similar to this issue with Telegraph: https://github.com/influxdata/telegraf/issues/10944

We have some applications that send the metrics to the agent using `micrometer-registry-statsd` and would like to have the same behavior when sending with `micrometer-registry-datadog`

**Comment by jon-signal** [comment]
We've also just run into this issue. In our use case, we switched from using `micrometer-registry-datadog` to `micrometer-registry-statsd` and wound up with some very surprising results where our count/rate metrics appeared to plummet by a factor of 10. Eventually, the axes on our charts self-corrected, but there's a factor-of-10 discontinuity that I don't think we'll be able to repair. I recognize that the presentation of our metrics within Datadog is beyond Micrometer's control, but even so, I think there are things that can be done to address the situation from Micrometer's end.

To elaborate on the cause, a Datadog support engineer granted me permission to quote his explanation:

> Moving from `api.Metric.send` (from Micrometer API) to `dog.count(...)` - it seems like it caused a type mismatch. The historical data was pre-aggregated (prior to submission) as a `COUNT` previously via the API. Like so:
>
> - [1,1,1,2,2,2,3,3] (10 second period)
> - COUNT = 15
> - RATE = 1.5
> 
> However, by moving to dogstatsd, this changed to:
>  
> - [1,1,1,2,2,2,3,3] (10 second period)
> - RATE= 15 / 10 = 1.5
> - COUNT= 1.5
>  
> By moving to the rate default, the data is being aggregated differently prior to submission. Discussed [here](https://docs.datadoghq.com/metrics/custom_metrics/type_modifiers/?tab=rate#in-application-modifiers):
> 
>> COUNT
>>
>> Effect of as_rate():
>>
>> Divides the result post-aggregation by the sampling interval in order to normalize it. For example, the following points submitted every second [1,1,1,1].as_rate() with a rollup interval of 20s would produce [0.05, 0.05, 0.05, 0.05].
>  
> So before the count was 15, but now due to the changes, the counts were made to be divided by the post-aggregation, making them to 1.5. Lowering the volumes by 10.
>  
> The old data is now showing with a rate: RATE = 15
>  
> As it took the '15' count and swapped that data to rate.

All the same, the Datadog support engineer notes:

> It just so happens that micrometer uses an API type that does not transition well to DogstatsD.

In short, I think the issue summary here is "switching from `micrometer-registry-datadog` to `micrometer-registry-statsd` can cause unexpected discontinuities in metrics." I won't prescribe a solution, but seems like it would be wise to at least document the risk somewhere in https://github.com/micrometer-metrics/micrometer-docs/blob/main/src/docs/implementations/datadog.adoc.

**Comment by shakuzen** [comment]
Sorry this sat waiting for triage for so long. I'm trying to go through old issues that seem to have been missed.

> Send metrics to the Datadog API with the same format the datadog agent uses. More specifically sending counts as rate metrics.

Making that switch would be a breaking change I guess, but it sounds like something we could make configurable if there's still interest in it. Would anyone be willing to make a pull request to make it configurable? I wonder if it makes sense to make it configurable for all counts in the registry or should it be configurable on a per meter basis?
